src/providers/schema.ts

```diff
@@ -52,14 +52,21 @@ export default class DatabaseTreeProvider implements vscode.TreeDataProvider<Nod
                             console.log("SchemaReader: ", schema, schemas, schemaName);
 
                             if (schema) {
+                                const sourceCount = await this.getSources(schema.id).then(s => s.length);
```

Comment:
What if, instead of using multiple `await` statements for individual queries, we use a single query to retrieve all the counts, or put all the promises in one array and execute them in parallel?

E.g.:

```javascript
const promises = [];
promises.push(this.getSources(schema.id).then(s => s.length));
...

const [sourceCount, ...] = await Promise.all(promises);
```

Reply:
Good idea! Just refactored this as requested.